<launch>

   <include file="$(find openni_launch)/launch/openni.launch">
  </include>
  
  <param name="/camera/driver/depth_registration" value="true"/>

  <node name="camera_tracker" pkg="dvo_ros" type="camera_tracker" respawn="false" output="screen">
  </node>

<param name="/camera_tracker/reconstruction" value="true"/>
<param name="/camera_tracker/use_weighting" value="true"/>
<param name="/camera_tracker/run_dense_tracking" value="true"/>
<param name="/camera_tracker/use_dense_tracking_estimate" value="true"/>

       <node name="omnimapper_ros_node" pkg="omnimapper_ros" type="omnimapper_ros_node" respawn="false" output="screen">
      	    <param name="use_planes" value="false"/>
	    <param name="use_icp" value="true"/>
	    <param name="use_objects" value="false"/>
	    <param name="use_csm" value="false"/>
	    <param name="use_occ_edge_icp" value="false"/>
	    <param name="use_tf" value="true"/>
	    <param name="use_error_plugin" value="false"/>
	    <param name="use_tsdf_plugin" value="false"/>
	    <param name="use_no_motion" value="false"/>
	    <param name="use_rgbd_sensor_base_tf_functor" value="false"/>
	    <param name="use_distortion_model" value="false"/>
      	    <param name="odom_frame_name" value="/world"/>
	    <param name="base_frame_name" value="/base_link_estimate"/>
	    <param name="use_init_pose" value="false"/>
	    <param name="init_pose_from_tf" value="true"/>
	    <param name="init_x" value="1.314809"/>
	    <param name="init_y" value="0.847662"/>
	    <param name="init_z" value="1.519455"/>
	    <param name="init_qx" value="0.886385"/>
	    <param name="init_qy" value="0.231053"/>
	    <param name="init_qz" value="-0.089359"/>
	    <param name="init_qw" value="-0.391090"/>
	    <param name="cloud_topic_name" value="/camera/depth_registered/points"/>
	    <param name="icp_leaf_size" value="0.025"/>
	    <param name="icp_max_correspondence_distance" value="0.1"/>
	    <param name="icp_trans_noise" value="0.05"/>
	    <param name="icp_rot_noise" value="0.05"/>
	    <param name="icp_add_loop_closures" value="false"/>
	    <param name="occ_edge_trans_noise" value="0.05"/>
	    <param name="occ_edge_rot_noise" value="0.1"/>
	    <param name="occ_edge_max_correspondence_dist" value="0.05"/>
	    <param name="occ_edge_score_thresh" value="0.1"/>
	    <param name="occ_edge_add_identity_on_fail" value="false"/>
	    <param name="tf_trans_noise" value="0.05"/>
	    <param name="tf_rot_noise" value="0.0872664626"/>
	    <param name="draw_pose_array" value="false"/>
	    <param name="add_pose_per_cloud" value="false"/>
	    <param name="broadcast_current_pose" value="true"/>
	    <param name="broadcast_map_to_odom" value="false"/>
	    <param name="evaluation_mode" value="false"/>
	    <param name="use_organized_feature_extraction" value="false"/>
	    <param name="use_rgbd_sensor_base_tf_functor" value="false"/>
	    <param name="evaluation_pcd_path" value="/home/atrevor/Desktop/out_pcds/rgb"/>
	    <param name="evaluation_associated_txt_path" value="/home/atrevor/Desktop/psf/Home/data/rgbd_dataset/rgbd_dataset_freiburg1_desk/associated.txt"/>
	    <param name="evaluation_ground_truth_txt_path" value="/home/atrevor/Desktop/psf/Home/data/rgbd_dataset/rgbd_dataset_freiburg1_desk/groundtruth.txt"/>
	    <param name="evaluation_output_trajectory_txt_path" value="/home/atrevor/Desktop/out_trajectory.txt"/>
      </node>

    <!--   <node name="oculus_to_camera_link" pkg="tf" type="static_transform_publisher" args="0.0 0.0 0.0 -0.5 0.5 -0.5 0.5 /base_link_estimate /camera_link 100" respawn="false" output="screen"/> -->
<!--        <node name="oculus_to_camera_link" pkg="tf" type="static_transform_publisher" args="0.0 0.0 0.0 -0.5 0.5 -0.5 0.5 /current_pose /camera_link 100" respawn="false" output="screen"/> -->
        <node name="oculus_to_camera_link" pkg="tf" type="static_transform_publisher" args="0.0 0.0 0.0 0.0 0.0 0.0 1.0 /base_link_estimate /camera_link 100" respawn="false" output="screen"/>  
</launch>